pkg/hintrunner/zero/zerohint_dictionaries.go

```diff
@@ -126,6 +126,86 @@ func createDictReadHinter(resolver hintReferenceResolver) (hinter.Hinter, error)
 	return newDictReadHint(dictPtr, key, value), nil
 }
 
+// DictWrite hint updates the value of given key for given dictionary
+// while also writing the prev_value of the key to memory
+//
+// `newDictWriteHint` takes 3 operander as argument
+//   - `dictPtr` variable will be pointer to the dictionary to update
+//   - `key` variable will be the key whose value is updated in the dictionary
+//   - `newValue` variable will be the new value for given `key` in the dictionary
+func newDictWriteHint(dictPtr, key, newValue hinter.ResOperander) hinter.Hinter {
+	return &GenericZeroHinter{
+		Name: "DictWrite",
+		Op: func(vm *VM.VirtualMachine, ctx *hinter.HintRunnerContext) error {
+			//> dict_tracker = __dict_manager.get_tracker(ids.dict_ptr)
+			//> dict_tracker.current_ptr += ids.DictAccess.SIZE
+			//> ids.dict_ptr.prev_value = dict_tracker.data[ids.key]
+			//> dict_tracker.data[ids.key] = ids.new_value
+
+			//> dict_tracker = __dict_manager.get_tracker(ids.dict_ptr)
+			dictPtr, err := hinter.ResolveAsAddress(vm, dictPtr)
+			if err != nil {
+				return err
+			}
+			dictionaryManager, ok := ctx.ScopeManager.GetZeroDictionaryManager()
+			if !ok {
+				return fmt.Errorf("__dict_manager not in scope")
+			}
+
+			//> dict_tracker.current_ptr += ids.DictAccess.SIZE
+			err = dictionaryManager.IncrementFreeOffset(*dictPtr, 3)
```

Comment:
same here, dont we need to write on the memory segment dedicated to this dict the entry created when writing in the dict?

Reply:
See: https://github.com/starkware-libs/cairo-lang/blob/efa9648f57568aad8f8a13fbf027d2de7c63c2c0/src/starkware/cairo/common/dict.cairo#L38

The `dict_write` cairo function does that work!

The `prev_value` is written to memory inside this hint though. See line `177`.

Reply:
Perfect !